pageserver/src/tenant/timeline.rs

```diff
@@ -922,14 +952,12 @@ impl Timeline {
             ),
         });
 
-        let gc_lock = self.layer_removal_cs.lock().await;
         let mut layers = self.layers.write().unwrap();
         let mut updates = layers.batch_update();
-        self.delete_historic_layer(&gc_lock, local_layer, &mut updates)?;
+        self.delete_historic_layer(layer_removal_cs, Arc::clone(local_layer), &mut updates)?;
         updates.insert_historic(new_remote_layer);
```

Comment:
If you rebase, you should see the `replace` now being used here in a conflict.

Reply:
Yeah, right now this branch targets #3552 but I think I'll spin it out once your #3513 is merged.